### PR TITLE
aria-selected added to Item Edit "Status" Tab. Fixed ID: 470076

### LINKED ISSUE #1190: [Deque Analysis] Item Edit "Status" Tab "serious" accessibility issues
> Serious: This issue results in serious barriers for people with disabilities, and will partially prevent them from accessing fundamental features or content. People relying on assistive technologies will experience significant frustration as a result. Issues falling under this category are major problems, and remediation should be a priority. 

## Deque Analysis Summary

Our Item Edit "Status" tab has a total of 5 serious accessibility issues (not including color contrast which are handled in other tickets).

1. ~~(IDs 470076, 471046, 471452) "Selected state of the element is missing or incorrect." on all the tabs (not just Status).  When a specific tab is active it should have `aria-current` or `aria-selected` specified to note the currently active page/tab.~~ FIXED by #2063 
    * ~~See also https://www.w3.org/TR/wai-aria-1.1/#aria-selected,  https://www.w3.org/TR/wai-aria-practices-1.1/#tabpanel and/or https://www.w3.org/TR/wai-aria-1.1/#aria-current~~
2. ~~(ID 470077) "Purpose of the link is not clear in context." on the "Withdraw..." and "Move..." buttons/links (on Status tab). The links should have more detail ("Withdraw this Item" "Move this Item to a different Collection") either in the text itself, or in an `aria-label`.~~ FIXED by #2062 
3. (ID 471072) Content overlap at 320px width. When the browser has 320px width, the entire page suffers from content overlap.

Full list of issues is viewable at (requires login): https://axeauditor.dequecloud.com/test-run/0856438a-a19a-11eb-bc31-b7d5be387c86/issues?activeTab=dt-issue&page=0&pageSize=50&sortField=ordinal&sortDir=asc&row=9&filter%5Bseverity%5D=3&filter%5Btype%5D=issue&filter%5Bpage_number%5D=12

## More Information / Tools

* Consider using [Deque's free Chrome plugin](https://chrome.google.com/webstore/detail/axe-devtools-web-accessib/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US) to check your work
* Deque's color contrast tool: https://dequeuniversity.com/rules/axe/4.1/color-contrast?application=axeAPI

### PR DESCRIPTION
**Pull request by CrisGuzmanS**
Hi @tdonohue ✋, I am submitting this pull request as a solution to activity ID 470076 in issue 1190. Regarding "Selected state of the element is missing or incorrect."

## References
* Fixes first issue noted in #1190

## Description
in each of the tabs of the page that opens when editing an item, an aria-selected has been added, if the tab is active, it will have the value of "true", if not, it will have the value of "false".

## Instructions for Reviewers

List of changes in this PR:
* _[attr.aria-selected]="page.page === currentPage"_ added in _edit-item-page.component.html_

To test my PR:
1. Login as adminsitrator
2. Edit any item
3. check the value of aria-selected of each li tag in the displayed tab.
image

## Checklist

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [ESLint](https://eslint.org/) validation using `yarn lint`
- [x] My PR doesn't introduce circular dependencies (verified via `yarn check-circ-deps`)
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new libraries/dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR includes new features or configurations, I've provided basic technical documentation in the PR itself.
- [x] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

### CODE REVIEW

File: src/app/item-page/edit-item-page/edit-item-page.component.html [hunk 1]
```diff
@@ -4,7 +4,7 @@
             <h2 class="border-bottom">{{'item.edit.head' | translate}}</h2>
             <div class="pt-2">
                 <ul class="nav nav-tabs justify-content-start">
-                    <li *ngFor="let page of pages" class="nav-item">
+                    <li *ngFor="let page of pages" class="nav-item" [attr.aria-selected]="page.page === currentPage">
```
Comment (tdonohue) [inline src/app/item-page/edit-item-page/edit-item-page.component.html:7]: I think this looks good!  However, I just realized that the recommendation for tabs is to set the roles "tablist" and "tab".  See the docs at https://www.w3.org/WAI/ARIA/apg/patterns/tabs/

So, I think we need to add two more small things here.  
* First, on the `<ul>` just above this we should have `<ul role="tablist"..`
* Seconding, on this `<li>` we should add the `<li role="tab"..`

Once those are in place, I think this aligns better with the recommendations in the docs above.

**Review by tdonohue — COMMENTED** [review]
@CrisGuzmanS : Thanks again!  This looks good & it works.  However, I realized my recommendations were missing the Aria "role" attributes that I believe these tabs still require.  If you could add them in this PR as well, that'd be appreciated.  See my inline comment below.

(If you don't have time to make these small changes, let me know. We can always move this PR forward as-is and see if someone else can add the "role" attributes)

**Comment by CrisGuzmanS** [comment]
> @CrisGuzmanS : Thanks again! This looks good & it works. However, I realized my recommendations were missing the Aria "role" attributes that I believe these tabs still require. If you could add them in this PR as well, that'd be appreciated. See my inline comment below.
> 
> (If you don't have time to make these small changes, let me know. We can always move this PR forward as-is and see if someone else can add the "role" attributes)

@tdonohue i can work on it. Tomorrow i will update the code

**Comment by CrisGuzmanS** [comment]
@tdonohue i already did the changes you recommended me. Please, let me know if everything is ok 👍

**Review by tdonohue — APPROVED** [review]
👍 Thanks @CrisGuzmanS !  Finally got back to testing this & I can verify this fixes that first issue in #1190. Will merge immediately

**Pull request merged**